clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -620,20 +622,23 @@ public void startConsumption(
       Optional<LeaderFollowerStateType> leaderState) {
 
     final String topic = veniceStore.getStoreVersionName();
+
     try (AutoCloseableLock ignore = topicLockManager.getLockForResource(topic)) {
-      StoreIngestionTask consumerTask = topicNameToIngestionTaskMap.get(topic);
-      if (consumerTask == null || !consumerTask.isRunning()) {
-        consumerTask = createConsumerTask(veniceStore, partitionId);
-        topicNameToIngestionTaskMap.put(topic, consumerTask);
-        versionedIngestionStats.setIngestionTask(topic, consumerTask);
+      StoreIngestionTask storeIngestionTask = topicNameToIngestionTaskMap.get(topic);
```

Comment:
`topicNameToIngestionTaskMap` is a concurrent map and technically this kind of get() + null check is not thread safe. Is it possible to replace with `compute` or `computeIfAbsent`.

Reply:
In theory, yes, but it is also guarded by the external topic lock though, so in theory it is already ok..

Reply:
Although I can still do the update, which has no more harm

Reply:
Let me see if it is easy to change it. Will update the PR

Reply:
Fixed.